src/cryptsetup/cryptsetup.c

```diff
@@ -565,11 +584,96 @@ static char *friendly_disk_name(const char *src, const char *vol) {
         return name_buffer;
 }
 
+static PasswordRegisteredType check_registered_passwords(struct crypt_device *cd) {
+
+        int slots = 0, slot_max;
+        PasswordRegisteredType r = _PASSWORD_REGISTERED_TYPE_INVALID;
+
+        assert(cd);
+
+        if (!streq_ptr(crypt_get_type(cd), CRYPT_LUKS2)) {
+                log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Non-LUKS2 devices only support passphrase.");
```

Comment:
Please drop the errno here, just use `log_debug()`. And it'd be better to say something like
```C
log_debug("%s: not a LUKS2 device, only passphrases are supported", pretty);
```
(`pretty` can be passed in by the caller.)

Reply:
Would it be enough to use crypt_get_device_name(cd)?